### PR TITLE
Misc file and docs fixes

### PR DESCRIPTION
**Pull request by abidlabs**
A grab bag PR to fix some small issues:

* Updates gradio's `requirements.txt` to requires `uvicorn>=0.14.0` (released May 28, 2021 so it should hopefully not cause any dependency conflicts)

Fixes: #4000, Fixes: #4065

* Updates error messaging to say: "when localhost is not accessible, a shareable link must be created. Please set share=True Please set share=True **or check your proxy settings to allow access to localhost**."

Closes: #4046

* Renames simplified Chinese translation file from `zh-cn.json` to `zh-CN.json` so that it actually works

image

Closes: #3870

* Documents iframe permissions in the sharing Guide

Closes: #3070 

* Improves the documentation for custom css / JS, adding appropriate warnings to use sparingly.

Closes: #3745

* Adds examples of more complex Chatbots to the Chatbot Guide

Closes: #3849

* Major cleanups to this Guide: https://gradio.app/using-hugging-face-integrations (was very out of date)

Closes: #3199

* Adds a section on Colab notebooks:

Closes: https://github.com/gradio-app/gradio/issues/4077

### CODE REVIEW

File: guides/04_integrating-other-frameworks/01_using-hugging-face-integrations.md [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
+Contributed by <a href="https://huggingface.co/osanseviero">Omar Sanseviero</a> 🦙 
 
 ## Introduction
 
-The Hugging Face Hub is a central platform that has over 90,000 [models](https://huggingface.co/models), 14,000 [datasets](https://huggingface.co/datasets) and 14,000 [demos](https://huggingface.co/spaces), also known as Spaces. From Natural Language Processing to Computer Vision and Speech, the Hub supports multiple domains. Although Hugging Face is famous for its 🤗 transformers and diffusers libraries, the Hub also supports dozens of ML libraries, such as PyTorch, TensorFlow, spaCy, and many others.
+The Hugging Face Hub is a central platform that has over 190,000 [models](https://huggingface.co/models), 32,000 [datasets](https://huggingface.co/datasets) and 40,000 [demos](https://huggingface.co/spaces), also known as Spaces. Although Hugging Face is famous for its 🤗 transformers and diffusers libraries, the Hub also supports dozens of ML libraries, such as PyTorch, TensorFlow, spaCy, and many others across a variety of domains, from computer vision to reinforcment learning.
```
Comment (abidlabs) [inline guides/04_integrating-other-frameworks/01_using-hugging-face-integrations.md:10]:
```suggestion
The Hugging Face Hub is a central platform that has over 190,000 [models](https://huggingface.co/models), 32,000 [datasets](https://huggingface.co/datasets) and 40,000 [demos](https://huggingface.co/spaces), also known as Spaces. Although Hugging Face is famous for its 🤗 transformers and diffusers libraries, the Hub also supports dozens of ML libraries, such as PyTorch, TensorFlow, spaCy, and many others across a variety of domains, from computer vision to reinforcement learning.
```

File: guides/04_integrating-other-frameworks/01_using-hugging-face-integrations.md [hunk 2]
```diff
@@ -164,25 +101,42 @@ file_url = upload_file(
...
 
 
 ## Embedding your Space demo on other websites
 
-Throughout this guide, you've seen there are Gradio demos embedded. You can also do this on own website! The first step is to create a Space with the demo you want to showcase. You can embed it in your HTML code, as shown in the following self-contained example.
+Throughout this guide, you've seen many embedded Gradio demos embedded. You can also do this on own website! The first step is to create a Hugging Face Space with the demo you want to showcase. Then, [follow the steps here to embed the Space on your website](/sharing-your-app/#embedding-hosted-spaces).
```
Comment (abidlabs) [inline guides/04_integrating-other-frameworks/01_using-hugging-face-integrations.md:108]:
```suggestion
Throughout this guide, you've seen many embedded Gradio demos. You can also do this on own website! The first step is to create a Hugging Face Space with the demo you want to showcase. Then, [follow the steps here to embed the Space on your website](/sharing-your-app/#embedding-hosted-spaces).
```

File: guides/01_getting-started/03_sharing-your-app.md [hunk 3]
```diff
@@ -113,10 +113,10 @@ To embed with IFrames instead (if you cannot add javascript to your website, for
 
 Again, you can find the `src=` attribute to your Space's embed URL, which you can find in the "Embed this Space" button.
 
-You'll also need to add a fixed `height` manually as well as other regular iframe attributes. For example: 
+You'll also need to add a fixed `height` manually as well as set the `frameborder`. In addition, if your app requires permissions such as access to the webcam or the microphone, you'll need to provide that as well using the `allow` attribute. Here's an example of how it might look:
```
Comment (aliabid94) [inline guides/01_getting-started/03_sharing-your-app.md:116]: Why did you add the frameborder note? It is a deprecated parameter according to https://developer.mozilla.org/en-US/docs/Web/HTML/Element/iframe#deprecated_attributes, should use regular CSS border controls instead.

File: guides/01_getting-started/03_sharing-your-app.md [hunk 4]
```diff
@@ -113,10 +113,10 @@ To embed with IFrames instead (if you cannot add javascript to your website, for
 
 Again, you can find the `src=` attribute to your Space's embed URL, which you can find in the "Embed this Space" button.
 
-You'll also need to add a fixed `height` manually as well as other regular iframe attributes. For example: 
+You'll also need to add a fixed `height` manually as well as set the `frameborder`. In addition, if your app requires permissions such as access to the webcam or the microphone, you'll need to provide that as well using the `allow` attribute. Here's an example of how it might look: 
 
 ```html
-&lt;iframe src="https://abidlabs-pytorch-image-classifier.hf.space" frameBorder="0" height="900">&lt;/iframe>
+&lt;iframe src="https://abidlabs-pytorch-image-classifier.hf.space" frameBorder="0" height="900" allow="camera; document-domain; legacy-image-formats; microphone; picture-in-picture;">&lt;/iframe>
```
Comment (aliabid94) [inline guides/01_getting-started/03_sharing-your-app.md:119]: can we remove `document-domain; legacy-image-formats;` allows? any need for them?

File: guides/02_building-interfaces/04_advanced-interface-features.md [hunk 5]
```diff
@@ -41,6 +41,8 @@ If you'd like to reference external files in your css, preface the file path (wh
 gr.Interface(..., css="body {background-image: url('file=clouds.jpg')}")
 ```
 
+**Warning**: Custom CSS is *not* guaranteed to work across Gradio versions as the Gradio HTML DOM may change. We recommend using custom CSS sparingly.
```
Comment (aliabid94) [inline guides/02_building-interfaces/04_advanced-interface-features.md:44]: Encourage using themes instead, which are supported?

File: guides/03_building-with-blocks/04_custom-CSS-and-JS.md [hunk 6]
```diff
@@ -2,6 +2,8 @@
 
 This guide covers how to style Blocks with more flexibility, as well as adding Javascript code to event listeners. 
 
+**Warning**: Custom JS and CSS is *not* guaranteed to work across Gradio versions as the Gradio HTML DOM may change. We recommend using custom CSS and JS sparingly.
```
Comment (aliabid94) [inline guides/03_building-with-blocks/04_custom-CSS-and-JS.md:5]: JS should work across gradio versions - it's specifically query selectors that may break.
```suggestion
**Warning**: The use of query selectors in custom JS and CSS is *not* guaranteed to work across Gradio versions as the Gradio HTML DOM may change. We recommend using query selectors sparingly.

File: guides/04_integrating-other-frameworks/01_using-hugging-face-integrations.md [hunk 7]
```diff
@@ -27,21 +27,32 @@ pipe = pipeline("translation", model="Helsinki-NLP/opus-mt-en-es")
...
 
-This demo requires installing four libraries: gradio, torch, transformers, and sentencepiece. Apart from that, this is a Gradio with the structure you're used to! The demo is a usual Gradio `Interface` with a prediction function, a specified input, and a specified output. The prediction function executes the `pipeline` function with the given input, retrieves the first (and only) translation result, and returns the `translation_text` field, which you're interested in.
+<gradio-app src="https://gradio-helsinki-translation-en-es.hf.space"></gradio-app>
```
Comment (aliabid94) [inline guides/04_integrating-other-frameworks/01_using-hugging-face-integrations.md:54]: why not use the `space=` attribute instead, to get the gradio spaces footer.

**Review by aliabid94 — APPROVED** [review]

File: guides/01_getting-started/03_sharing-your-app.md (hunk 3 above)
Comment (abidlabs) [reply guides/01_getting-started/03_sharing-your-app.md:116]: Cool, good to know. Will replace with `style="border:0;"` because otherwise the ugly border pops up.

File: guides/01_getting-started/03_sharing-your-app.md (hunk 4 above)
Comment (abidlabs) [reply guides/01_getting-started/03_sharing-your-app.md:119]: Yup that should be fine

File: guides/02_building-interfaces/04_advanced-interface-features.md (hunk 5 above)
Comment (abidlabs) [reply guides/02_building-interfaces/04_advanced-interface-features.md:44]: Sounds good I'll encourage themes

**Comment by abidlabs** [comment]
Thanks @aliabid94 for the review! Have address your comments and will merge once CI runs
